src/App.tsx

```diff
@@ -1,60 +1,109 @@
-import classNames from 'classnames';
-
+/* eslint-disable @typescript-eslint/indent */
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import cn from 'classnames';
 import './App.scss';
 
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
-
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+import { PostsList, PostDetails, UserSelector, Loader } from './components';
+import { User } from './types';
+import { usePosts, useUsers } from './hooks';
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+export const App = () => {
+  const {
+    users,
+    errorMessage: usersErrorMessage,
+    selectedUser,
+    setSelectedUser,
+  } = useUsers();
+  const {
+    posts,
+    selectedPost,
+    setSelectedPost,
+    isLoading,
+    setIsLoading,
+    errorMessage: postsErrorMessage,
+  } = usePosts(selectedUser?.id);
 
-              <Loader />
+  const errorMessage = usersErrorMessage || postsErrorMessage;
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const handleSelectUserId = (user: User) => {
+    setIsLoading(true);
+    setSelectedPost(null);
+    setSelectedUser(user);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    if (selectedUser === user) {
+      setIsLoading(false);
+    }
```

Comment:
The condition `if (selectedUser === user)` is used to set `setIsLoading(false)`. This might not be necessary since `setIsLoading(true)` is called right before setting the selected user. Consider reviewing this logic to ensure it behaves as expected.